lib/fs.js

```diff
@@ -2034,6 +2034,11 @@ ReadStream.prototype.close = function(cb) {
   this.fd = null;
 };
 
+// needed because as it will be called with arguments
+// that does not match this.close() signature
+function closeOnOpen() {
```

Comment:
Maybe we can make it more evident by adding an `fd` parameter here? Plus I'm sure V8 would prefer it.